tests/Backend/Mixed/SharingTest.php

```diff
@@ -776,118 +776,61 @@ public function testSharedBuckets($backend, $isAsync): void
 
-        // search from other project in same organization > should return same results
-        // created 6 buckets 2 in "otherOrg" and 2 shared buckets, 2 buckets from main project and 4 buckets from linked project should be searchable
-        $apiCall3 = fn() => $this->_client2->globalSearch($bucketNameHash);
-        $assertCallback3 = function ($searchResult) {
-            $this->assertSame(6, $searchResult['all']);
+        // test can find bucket in both projects
+        $apiCall = fn() => $this->_client->globalSearch($bucketNameHash);
+        $assertCallback = function ($searchResult) {
+            $this->assertSame(3, $searchResult['all']);
             $this->assertSame('bucket', $searchResult['items'][0]['type']);
             $this->assertCount(2, $searchResult['byProject']);
         };
-        $this->retryWithCallback($apiCall3, $assertCallback3);
+        $this->retryWithCallback($apiCall, $assertCallback);
 
-        // search from anotherOrg - only buckets in another org should be searchable
-        $apiCall4 = fn() => $this->clientAdmin3InOtherOrg->globalSearch($bucketNameHash);
-        $assertCallback4 = function ($searchResult) {
-            $this->assertSame(2, $searchResult['all']);
+        // test linked bucket can be found in linking project
+        $apiCall = fn() => $this->_client->globalSearch($bucketNameHash, new GlobalSearchOptions(null, null, null, [$projectLinking]));
+        $assertCallback = function ($searchResult) {
+            $this->assertSame(1, $searchResult['all']);
             $this->assertSame('bucket', $searchResult['items'][0]['type']);
-            $this->assertCount(1, $searchResult['byProject']);
+            $this->assertCount(2, $searchResult['byProject']);// this is count in both projects
         };
-        $this->retryWithCallback($apiCall4, $assertCallback4);
-
-        // POST clean-up
-
-        // drop shared first
-        /** @var Client $client */
-        foreach ($clients as $client) {
-            foreach ($client->listBuckets() as $bucket) {
-                if (!empty($bucket['sourceBucket'])) {
-                    $client->dropBucket($bucket['id']);
-                }
-            }
-        }
+        $this->retryWithCallback($apiCall, $assertCallback);
 
-        /** @var Client $client */
-        foreach ($clients as $client) {
-            foreach ($client->listBuckets() as $bucket) {
-                $client->dropBucket($bucket['id']);
-            }
-        }
+        // in project in other organization can't find nothing
+        $apiCallOtherOrg = fn() => $this->clientAdmin3InOtherOrg->globalSearch($bucketNameHash);
+        $assertCallback4 = function ($searchResult) {
+            $this->assertSame(0, $searchResult['all']);
+            $this->assertCount(0, $searchResult['byProject']);
+        };
+        $this->retryWithCallback($apiCallOtherOrg, $assertCallback4);
```

Comment:
otestovat, ze na to nevidi iny projekt v inej org staci takto, ze nevidim proste nic naindexovane ked som mu nevytvaral buckety. Vytvarat nejake buckety v tom tretom projekte imho neni treba testovali by sme zas len to ze sa buckety zaindexeuju a to mame testovane uz na viacerych miestach